nuxeo-jsf/nuxeo-platform-actions-jsf/src/main/java/org/nuxeo/ecm/platform/actions/jsf/JSFActionContext.java

```diff
@@ -91,13 +115,26 @@ public JSFActionContext(ELContext originalContext, ExpressionFactory expressionF
         putLocalVariable("SeamContext", new SeamContextHelper());
 
         // evaluate expression
-        ValueExpression ve = expressionFactory.createValueExpression(finalContext, expr, Boolean.class);
-        return (T) ve.getValue(finalContext);
+        ValueExpression valueExpression = expressionFactory.createValueExpression(finalContext, expr, expectedType);
+        Class<?> type = valueExpression.getType(originalContext);
+        boolean valid = type != null && ClassUtils.isAssignable(type, expectedType);
+        return new ValueExpressionDetails(finalContext, valueExpression, valid);
     }
 
-    @Override
-    public boolean checkCondition(String expression) throws ELException {
-        return Boolean.TRUE.equals(evalExpression(expression, Boolean.class));
+    protected static class ValueExpressionDetails {
```

Comment:
I would also change `ValueExpression` -> `Expression` everywhere (new class and methods)